classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -377,12 +407,23 @@ private function _migrateSubmissionFiles() {
 				'path' => $path,
 				'mimetype' => $row->file_type,
 			], 'file_id');
+			$fileId = $fileIdMap[$row->file_id] ?? $row->file_id;
+			$updateData = ['new_file_id' => $newFileId];
+			// If the file_id was forked into a new one, we've got to update the references at the review_round_files and the submission_file itself
+			if ($fileId !== $row->file_id) {
+				$updateData['file_id'] = $fileId;
+				$updateData['source_file_id'] = $fileIdMap['sourceFileId'];
+				Capsule::table('review_round_files')
+					->where('file_id', '=', $row->file_id)
+					->where('revision', '=', $row->revision)
+					->update(['file_id' => $fileId]);
```

Comment:
Here I decoupled the `review_round_files`, given that some lines below it will be "consolidated" by `file_id`.